### PR TITLE
Bugfix/issue 2562 - RequestBody as Operation property.

### PR DESCRIPTION
**Pull request by ralphdoe**
Reader Changes
Reader Tests Implemented

### CODE REVIEW

File: modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java [hunk 1]
```diff
@@ -449,6 +448,14 @@ public OpenAPI read(Class<?> cls, String parentPath) {
...
 
                     openAPI.setPaths(this.paths);
+
+                    io.swagger.v3.oas.annotations.Operation apiOperation =
+                            ReflectionUtils.getAnnotation(method, io.swagger.v3.oas.annotations.Operation.class);
+                    // requestBody
+                    if (apiOperation != null && apiOperation.requestBody() != null && operation.getRequestBody() == null) {
+                        OperationParser.getRequestBody(apiOperation.requestBody(), classConsumes, methodConsumes, components).ifPresent(
+                                requestBodyObject -> operation.setRequestBody(requestBodyObject));
+                    }
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java]: This doesn't seem to handle `@RequestBody` as method level annotation

File: modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/resources/RequestBodyResource.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @POST
+    @Operation(summary = "Create user",
+            description = "This can only be done by the logged in user.",
+            requestBody = @RequestBody(description = "RequestBody on Annotation", required = true,
+                    content = @Content(schema = @Schema(implementation = User.class))))
+    public Response createUser(
+            @Parameter(description = "Request Body in Parameter", required = true) User user) {
+        return Response.ok().entity("").build();
+    }
+}
```
Comment (frantuma) [inline modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/resources/RequestBodyResource.java]: Use existing `RequestBodyTest`, considering all scenarios (method annotation only, @Operation.requestBody annotation only, both, both and also parameter annotation)

File: modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java [hunk 3]
```diff
@@ -30,16 +31,16 @@
 /**
  * The annotation may be used to define a resource method as an OpenAPI Operation, and/or to define additional
  * properties for the Operation.
- *
+ * <p>
```
[the remark this answers is not in the document]
Comment (ralphdoe) [reply modules/swagger-annotations/src/main/java/io/swagger/v3/oas/annotations/Operation.java]: Added when reformatting code. Updated.

File: modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java (hunk 1 above)
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java]: Sure. Now I have added this functionality.

File: modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java [hunk 4]
```diff
@@ -711,7 +719,8 @@ private Operation parseMethod(
 
         // apiResponses
         if (apiResponses != null) {
-            OperationParser.getApiResponses(apiResponses.toArray(new io.swagger.v3.oas.annotations.responses.ApiResponse[apiResponses.size()]), classProduces, methodProduces, components).ifPresent(responses -> {
+            OperationParser.getApiResponses(apiResponses.toArray(new io.swagger.v3.oas.annotations.responses.ApiResponse[apiResponses.size()]),
+                    classProduces, methodProduces, components).ifPresent(responses -> {
```
[the remark this answers is not in the document]
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/Reader.java]: The line is so long. Divided in two lines.

File: modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/resources/RequestBodyResource.java (hunk 2 above)
Comment (ralphdoe) [reply modules/swagger-jaxrs2/src/test/java/io/swagger/v3/jaxrs2/resources/RequestBodyResource.java]: Sure. Added.

**Pull request merged**